src/main/scala/org/neo4j/spark/Neo4jOptions.scala

```diff
@@ -24,78 +29,165 @@ class Neo4jOptions(private val parameters: java.util.Map[String, String]) {
+
+    if (lifetime > -1) builder.withMaxConnectionLifetime(lifetime, TimeUnit.MILLISECONDS)
+    if (acquisitionTimeout > -1) builder.withConnectionAcquisitionTimeout(acquisitionTimeout, TimeUnit.MILLISECONDS)
+    if (livenessCheckTimeout > -1) builder.withConnectionLivenessCheckTimeout(livenessCheckTimeout, TimeUnit.MILLISECONDS)
+    if (connectionTimeout > -1) builder.withConnectionTimeout(connectionTimeout, TimeUnit.MILLISECONDS)
+
+    URI.create(url).getScheme match {
+      case "neo4j+s" | "neo4j+ssc" | "bolt+s" | "bolt+ssc" => Unit
+      case _ => if (!encryption) {
+        builder.withoutEncryption()
+      }
+      else {
+        builder.withEncryption()
+      }
+    }
+
+    trustStrategy
+      .map(Config.TrustStrategy.Strategy.valueOf)
+      .map {
+        case TrustStrategy.Strategy.TRUST_ALL_CERTIFICATES => TrustStrategy.trustAllCertificates()
+        case TrustStrategy.Strategy.TRUST_SYSTEM_CA_SIGNED_CERTIFICATES => TrustStrategy.trustSystemCertificates()
+        case TrustStrategy.Strategy.TRUST_CUSTOM_CA_SIGNED_CERTIFICATES => TrustStrategy.trustCustomCertificateSignedBy(new File(certificatePath))
+      }.foreach(builder.withTrustStrategy)
+
+    builder.build()
+  }
+
+  def toNeo4jAuth: AuthToken = {
+    auth match {
+      case "basic" => AuthTokens.basic(username, password)
+      case "none" => AuthTokens.none()
+      case "kerberos" => AuthTokens.kerberos(ticket)
+      case "custom" => AuthTokens.custom(principal, credentials, realm, schema)
+      case _ => throw new IllegalArgumentException(s"Authentication method '${auth}' is not supported")
+    }
+  }
+}
 
 object Neo4jOptions {
 
   // connection options
   val URL = "url"
-  val AUTH_TYPE = "authentication.type"
+
+  // auth
+  val AUTH_TYPE = "authentication.type" // basic, none, kerberos, custom
   val AUTH_BASIC_USERNAME = "authentication.basic.username"
   val AUTH_BASIC_PASSWORD = "authentication.basic.password"
+  val AUTH_KERBEROS_TICKET = "authentication.kerberos.ticket"
+  val AUTH_CUSTOM_PRINCIPAL = "authentication.custom.principal"
+  val AUTH_CUSTOM_CREDENTIALS = "authentication.custom.credentials"
+  val AUTH_CUSTOM_REALM = "authentication.custom.realm"
+  val AUTH_CUSTOM_SCHEME = "authentication.custom.scheme"
+
+  // driver
   val ENCRYPTION_ENABLED = "encryption.enabled"
   val ENCRYPTION_TRUST_STRATEGY = "encryption.trust.strategy"
   val ENCRYPTION_CA_CERTIFICATE_PATH = "encryption.ca.certificate.path"
-  val MAX_CONNECTION_LIFETIME_MSECS = "connection.max.lifetime.msecs"
-  val MAX_CONNECTION_ACQUISITION_TIMEOUT_MSECS = "connection.acquisition.timeout.msecs"
+  val CONNECTION_MAX_LIFETIME_MSECS = "connection.max.lifetime.msecs"
+  val CONNECTION_LIVENESS_CHECK_TIMEOUT = "connection.liveness.timeout.msec"
+  val CONNECTION_ACQUISITION_TIMEOUT_MSECS = "connection.acquisition.timeout.msecs"
+  val CONNECTION_TIMEOUT_MSECS = "connection.timeout.msecs"
 
   // session options
   val DATABASE = "database"
```

Comment:
I tested it, and get a NullPointerException if Database is specified as None in python.  If it's not a valid value, ignore the setting instead and user gets default database.

Reply:
can you share the code and the stacktrace?